From 7c98dc897cce44dcf71327135c01ed980cd4f185 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Mon, 17 Feb 2020 17:22:32 +0800
Subject: [PATCH] distros/rhel.py: changes for wrl

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 cloudinit/distros/rhel.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cloudinit/distros/rhel.py b/cloudinit/distros/rhel.py
index 1fecb61..e36151e 100644
--- a/cloudinit/distros/rhel.py
+++ b/cloudinit/distros/rhel.py
@@ -39,6 +39,7 @@ class Distro(distros.Distro):
     resolve_conf_fn = "/etc/resolv.conf"
     tz_local_fn = "/etc/localtime"
     usr_lib_exec = "/usr/libexec"
+    init_cmd = ['systemctl']
 
     def __init__(self, name, cfg, paths):
         distros.Distro.__init__(self, name, cfg, paths)
@@ -47,7 +48,7 @@ class Distro(distros.Distro):
         # should only happen say once per instance...)
         self._runner = helpers.Runners(paths)
         self.osfamily = 'redhat'
-        cfg['ssh_svcname'] = 'sshd'
+        cfg['ssh_svcname'] = 'sshd.socket'
 
     def install_packages(self, pkglist):
         self.package_command('install', pkgs=pkglist)
-- 
2.7.4

